src/NServiceBus.Core/Sagas/SagaPropertyMapper.cs

```diff
@@ -52,6 +52,21 @@ public IToSagaExpression<TSagaData> ConfigureHeaderMapping<TMessage>(string head
             return new MessageHeaderToSagaExpression<TSagaData, TMessage>(sagaHeaderFindingConfiguration, headerName);
         }
 
+        /// <summary>
+        /// Specify the correlation property for instances of <typeparamref name="TSagaData"/>.
+        /// </summary>
+        /// <param name="sagaProperty">The correlation property to use when finding a saga instance.</param>
+        /// <returns>
+        /// A <see cref="CorrelatedSagaPropertyMapper{TSagaData}"/> that provides the fluent chained
+        /// <see cref="CorrelatedSagaPropertyMapper{TSagaData}.ToMessage{TMessage}"/> to map a message type to
+        /// the correlation property.
+        /// </returns>
+        public CorrelatedSagaPropertyMapper<TSagaData> MapSaga(Expression<Func<TSagaData, object>> sagaProperty)
```

Comment:
Will the old way be obsoleted with warn to guide users?

Reply:
I am not sure. I am currently leaning towards waiting for v8 for that.

a) The new API is effectively syntactic sugar over the existing one. Even if we obsoleted the existing one, we'd still have one usage of it, which is this new API.
b) We'd need to change all of the tests in the core repo, and in all of the downstream repos
c) We'd need to change all of the docs samples and snippets

All of that is doable, but I'm wondering if an enhancement release is the best place to do it.